File: Python/recursion/tower_of_hanoi.py
def tower_of_hanoi(n, source, helper, destination):
    if n == 0:
        return
    tower_of_hanoi(n-1, source, destination, helper)
    print(f"Disk {n} from rod {source} to rod {destination}")
    tower_of_hanoi(n-1, helper, source, destination)


tower_of_hanoi(3, 'A', 'B', 'C')

File: TypeScript/recursion/towerOfHanoi.ts
function towerOfHanoi(n: number, source: string, helper: string, destination: string): void{
    if (n === 0){
        return;
    }
    towerOfHanoi(n-1, source, destination, helper);
    console.log(`Disk ${n} from rod ${source} to rod ${destination}`);
    towerOfHanoi(n-1, helper, source, destination);
}

towerOfHanoi(3, 'A', 'B', 'C');
